### PR TITLE
Added in yolo script for Auto Annotation

### PR DESCRIPTION
**Pull request by benhoff**

### CODE REVIEW

**Comment by benhoff** [comment]
regarding the Codacy errors, there's an issue with the way things are implemented, you can't do multilevel method names. See here: https://github.com/opencv/cvat/issues/652

**Comment by nmanovic** [comment]
@benhoff , thanks for the great contribution. As always.

I have a couple of questions and requests:
- Could you please add information about the contribution to CHANGELOG.md file?
- Did you expect any problems with provided instructions for converting yolo to IR? For example, I tried 3 different version of TF (2.0, 1.15, 1.13) and only the 1.13 version of TF worked as expected. The same for networkx python package.
- Does your converted model can find something on the image: https://farm3.staticflickr.com/2372/2220900131_59625951f6_z.jpg? I cannot find an image where my converted model can detect something.

**Comment by benhoff** [comment]
I got some matches, probably not as many as there should be.
Screen Shot 2019-10-24 at 10 48 31 AM

There appear to be some issues with the accuracy of the converted model.

See: https://github.com/PINTO0309/OpenVINO-YoloV3/issues/32
and: https://github.com/PINTO0309/OpenVINO-YoloV3/issues/34
Intel link: https://software.intel.com/en-us/forums/computer-vision/topic/808504#comment-1938506

I'll add some notes about the conversion issues. Specifically NetworkX and Tensorflow. I was on tensorflow 1.12.3, but had similar issues on a different laptop that was on Tensorflow 2.0. The DLDT repo needs to restrict the dependencies!

See here for NetworkX info: https://github.com/opencv/dldt/issues/294

**Comment by nmanovic** [comment]
> I'll add some notes about the conversion issues. 

OK. I will wait. Please add a note to CHANGELOG.md. In general I really want to merge it ASAP because we had a lot of requests in the past for YOLO support.

**Comment by mingweihe** [comment]
Hi, @benhoff @nmanovic thanks for your contribution on this awesome project.
I recently starting work on this repo, I'm trying to apply my pre-trained yolov5(https://github.com/ultralytics/yolov5) model on the auto-annotation module.
Until now, I've converted the yolov5 model to onnx model, I also used the openvino API to convert the onnx model to openvino model, so I have the .bin .json. .xml .mapping files in hand.
What I'm trying to do is to create a python interpretation script similar to yours @benhoff , I adjusted the code, not matched layer name issue has been solved, while it seems that there exists another issue which is no boxes produced. (my original model has a very good mAP/accuracy performance)
Can you help solve this issue, could this be caused by source code changes since last year's PR? @benhoff

**Comment by mingweihe** [comment]
On the other hand, I found the codes changed a lot since I pulled it down a few days ago, may I happen to know the reason of it, is that because you are not willing to let the complex customized auto-annotation module exposed to new developers?

**Comment by nmanovic** [comment]
@mingweihe , we have changed the implementation of automatic annotation feature. It is open source and now it supports any DL frameworks out of the box. Please see how to enable the feature here: https://github.com/opencv/cvat/blob/develop/cvat/apps/documentation/installation.md#semi-automatic-and-automatic-annotation

Also you can find all DL models which are available for automatic annotation here: https://github.com/opencv/cvat/tree/develop/serverless

**Comment by mingweihe** [comment]
@nmanovic Thank you, serverless seems very cool! great job!

**Comment by wartek69** [comment]
Hi @mingweihe 
Did you find a way to use your custom pretrained yolov5 model for automatic annotation?
The links shared by @nmanovic  do not contain yolov5 functions if I'm not mistaken.
